source/common/protobuf/utility.cc

```diff
@@ -0,0 +1,66 @@
+#include "common/protobuf/utility.h"
+
+#include "spdlog/spdlog.h"
+
+namespace Envoy {
+
+MissingFieldException::MissingFieldException(const std::string& field_name,
+                                             const Protobuf::Message& message)
+    : EnvoyException(
+          fmt::format("Field '{}' is missing in: {}", field_name, message.DebugString())) {}
+
+Protobuf::Message* MessageUtil::getMutableMessage(Protobuf::Message& message,
+                                                  const std::string& field_name) {
+  const auto* reflection = message.GetReflection();
+  const auto* field = message.GetDescriptor()->FindFieldByName(field_name);
+  return reflection->MutableMessage(&message, field);
+}
+
+const Protobuf::Message* MessageUtil::getMessage(const Protobuf::Message& message,
+                                                 const std::string& field_name) {
+  const auto* reflection = message.GetReflection();
+  const auto* field = message.GetDescriptor()->FindFieldByName(field_name);
+  if (reflection->HasField(message, field)) {
+    return &reflection->GetMessage(message, field);
+  }
+  return nullptr;
+}
+
+uint32_t MessageUtil::getUInt32(const Protobuf::Message& message, const std::string& field_name,
+                                uint32_t default_value) {
+  const auto* sub_message = getMessage(message, field_name);
+  if (sub_message != nullptr) {
+    return dynamic_cast<const ProtobufWkt::UInt32Value&>(*sub_message).value();
```

Comment:
Question: The use of dynamic_cast<> throughout this PR seems like it could definitely result in nullptr and crash. Is an invalid dynamic_cast assumed to be a programming error internally? (Basically I assume it's never possible that user input could lead to an invalid dynamic_cast<>)?